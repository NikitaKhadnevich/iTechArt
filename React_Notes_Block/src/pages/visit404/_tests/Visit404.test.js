import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Visit404List from '../Visit404List';

configure({ adapter: new Adapter() });

describe('About Page', () => {
  it('Is it ErrorPage Exist', () => {
    const SharedWrapper = shallow(<Visit404List />);
    expect(SharedWrapper).toHaveLength(1);
  });
  it('Is it ErrorPage Snap', () => {
    const SharedWrapper = shallow(<Visit404List />);
    expect(SharedWrapper).toMatchSnapshot();
  });
  it('There is ErrorPage has Title', () => {
    const SharedWrapper = shallow(<Visit404List />);
    expect(
      SharedWrapper.find('WithStyles(ForwardRef(Typography))').first().text()
    ).toEqual('Something went wrong - 404');
  });
  it('There is ErrorPage has 1 folders', () => {
    const SharedWrapper = shallow(<Visit404List />);
    expect(
      SharedWrapper.find('WithStyles(ForwardRef(Typography))')
    ).toHaveLength(1);
  });
  it('are ErrorPage Folder type are Paragraph', () => {
    const SharedWrapper = shallow(<Visit404List />);
    expect(
      SharedWrapper.findWhere(
        (n) => n.type() === 'WithStyles(ForwardRef(Typography))'
      )
    );
  });
});
