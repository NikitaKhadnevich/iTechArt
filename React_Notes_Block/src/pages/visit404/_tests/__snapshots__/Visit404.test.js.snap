// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Page Is it ErrorPage Snap 1`] = `
<Styled(WithStyles(ForwardRef(Container)))>
  <Styled(WithStyles(ForwardRef(Container)))>
    <WithStyles(ForwardRef(Typography))
      align="center"
      component="h5"
      variant="h5"
    >
      Something went wrong - 404
    </WithStyles(ForwardRef(Typography))>
  </Styled(WithStyles(ForwardRef(Container)))>
</Styled(WithStyles(ForwardRef(Container)))>
`;
